spec/lib/tasks/link_kinnex_compound_samples_spec.rb

```diff
@@ -15,12 +15,12 @@
 
   let!(:samples) do
     [
-      Sample.create!(id_sample_lims: compound_sample_id, name: 'supplier_x', id_lims: 'Traction',
-                     accession_number: 'ACC123', uuid_sample_lims: 123_456, last_updated: '2012-03-11 10:22:42'),
-      Sample.create!(id_sample_lims: component_sample_id1, supplier_name: 'supplier_x', uuid_sample_lims: 123_457,
-                     last_updated: '2012-03-11 10:22:42', id_lims: 'SQSCP'),
-      Sample.create!(id_sample_lims: component_sample_id2, supplier_name: 'supplier_x', uuid_sample_lims: 123_458,
-                     last_updated: '2012-03-11 10:22:42', id_lims: 'SQSCP')
+      Sample.create!(id_sample_lims: compound_sample_id, name: 'SP09a', id_lims: 'Traction',
+                     uuid_sample_lims: 123_456, last_updated: '2012-03-11 10:22:42'),
+      Sample.create!(id_sample_lims: component_sample_id1, supplier_name: 'SP09a', uuid_sample_lims: 123_457,
+                     last_updated: '2012-03-11 10:22:42', id_lims: 'SQSCP', accession_number: 'ACC123'),
+      Sample.create!(id_sample_lims: component_sample_id2, supplier_name: 'SP09a', uuid_sample_lims: 123_458,
+                     last_updated: '2012-03-11 10:22:42', id_lims: 'SQSCP', accession_number: 'ACC124')
```

Comment:
Could we add some extra dummy samples (different supplier name, no accession_number etec) to test the filters are correct and we are not false matching samples.